catalog/core/catalog-core-actions/src/test/java/org/codice/ddf/catalog/actions/AbstractMetacardActionProviderTest.java

```diff
@@ -0,0 +1,327 @@
+            return title;
+        }
+
+        @Override
+        public String getDescription() {
+            return description;
+        }
+    }
+
+    private class MetacardActionProvider extends AbstractMetacardActionProvider {
+
+        protected MetacardActionProvider(String actionProviderId, String title,
+                String description) {
+            super(actionProviderId, title, description);
+        }
+
+        @Override
+        protected boolean canHandleMetacard(Metacard metacard) {
+            return super.canHandleMetacard(metacard);
+        }
+
+        @Override
+        protected Action createMetacardAction(String actionProviderId, String title,
+                String description, URL url) {
+            return null;
+        }
+
+        @Override
+        protected URL getMetacardActionUrl(String metacardSource, Metacard metacard)
+                throws IOException {
+            return new URL("https://localhost/action");
+        }
+    }
+
+    @Test(expected = IllegalArgumentException.class)
+    public void constructorWithNullActionProviderId() {
+        new MetacardActionProvider(null, TITLE, DESCRIPTION);
+    }
+
+    @Test(expected = IllegalArgumentException.class)
+    public void constructorWithBlankActionProviderId() {
+        new MetacardActionProvider("  ", TITLE, DESCRIPTION);
+    }
+
+    @Test(expected = IllegalArgumentException.class)
+    public void constructorWithNullTitle() {
+        new MetacardActionProvider(ACTION_ID, null, DESCRIPTION);
+    }
+
+    @Test(expected = IllegalArgumentException.class)
+    public void constructorWithNullDescription() {
+        new MetacardActionProvider(ACTION_ID, TITLE, null);
+    }
+
+    @Test
+    public void canHandleWithNull() {
+        MetacardActionProvider actionProvider = spy(new MetacardActionProvider(ACTION_ID,
+                TITLE,
+                DESCRIPTION));
+        assertThat(actionProvider.canHandle(null), is(false));
+        verify(actionProvider, never()).canHandleMetacard(any());
+    }
+
+    @Test
+    public void canHandleWithNonMetacard() {
+        MetacardActionProvider actionProvider = spy(new MetacardActionProvider(ACTION_ID,
+                TITLE,
+                DESCRIPTION));
+        assertThat(actionProvider.canHandle("blah"), is(false));
+        verify(actionProvider, never()).canHandleMetacard(any());
+    }
+
+    @Test
+    public void canHandleDelegatesToCanHandleMetacard() {
+        MetacardActionProvider actionProvider = spy(new MetacardActionProvider(ACTION_ID,
+                TITLE,
+                DESCRIPTION));
+        when(actionProvider.canHandleMetacard(metacard)).thenReturn(true);
+
+        assertThat(actionProvider.canHandle(metacard), is(true));
+        verify(actionProvider).canHandleMetacard(metacard);
+    }
+
+    @Test
+    public void getActionsWithNull() throws IOException {
+        MetacardActionProvider actionProvider = spy(new MetacardActionProvider(ACTION_ID,
```

Comment:
You could possible make actionProvider a class member variable or return it through a private method like `getActionProvider()`. Maintenance wise if `MetacardActionProvider` were to change every test would have to change as well.